Neos.ContentRepository.Core/Classes/Factory/CommandHooksFactoryDependencies.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Neos.ContentRepository package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\Factory;
+
+use Neos\ContentRepository\Core\CommandHandler\CommandHookInterface;
+use Neos\ContentRepository\Core\Projection\ContentGraph\ContentGraphReadModelInterface;
+use Neos\ContentRepository\Core\SharedModel\ContentRepository\ContentRepositoryId;
+
+/**
+ * @api for implementers of custom {@see CommandHookInterface}s
+ */
+final readonly class CommandHooksFactoryDependencies
```

Comment:
... and the NodeTypemanager would't harm here as well 😅 sorry :D 

I just thought: IF we want to use this for node creation handler or decisions based on the nodetype we would need to pass that.

in the end probably envething that the catchups get? (i picked there all the details that would be api on the cr)

```php
    public function __construct(
        public ContentRepositoryId $contentRepositoryId,
        // public ProjectionStateInterface $projectionState,
        public NodeTypeManager $nodeTypeManager,
        public ContentDimensionSourceInterface $contentDimensionSource,
        public InterDimensionalVariationGraph $variationGraph
    ) {
    }
```

this is not really fun :D And we probably have to pass the projection states as well 😅 Basically a full readonly content repository.

The catchups get a semi full content repository (with only their projections state) ...

If there is a simple way we see to combine id go that but otherwise lets just duplicate that here :)

Reply:
resolved with af193a57ebc9e9c3d01748de02675825476421a1

(we decided to skip `ProjectionStates` for now, because there are some issues to be resolved and we can always add those later)